main.go

```diff
@@ -84,48 +88,90 @@ func main() {
 
 	// Make waitgroup and channels to process objects
 	// tasks asynchronously
-	var wg sync.WaitGroup
-	cDone := make(chan bool)
-	wg.Add(len(filtered_res.Contents))
-	go func() {
-		// Process each s3 object
-		for _, item := range filtered_res.Contents {
-			o := item
-			go func() {
-				defer wg.Done()
-				if err := processObject(ctx, s3Cli, dwn, influxWriter, o); err != nil {
-					log.Error().Err(err).
-						Msg("Processing error")
-				}
-			}()
-		}
+	if len(unproc_csvs.Contents) > 0 {
+		var wg sync.WaitGroup
+		cDone := make(chan bool)
+		wg.Add(len(unproc_csvs.Contents))
+		go func() {
+			// Process each s3 object
+			for _, item := range unproc_csvs.Contents {
+				o := item
+				go func() {
+					defer wg.Done()
+					if err := processObject(ctx, dwn, upl, influxWriter, o); err != nil {
+						log.Error().Err(err).
+							Msg("Processing error")
+					}
+				}()
+			}
 
-		wg.Wait()
-		close(cDone)
-	}()
+			wg.Wait()
+			close(cDone)
+		}()
+		<-cDone
+	}
```

Comment:
Is there a reason why you're processing all files and then cleaning all files ? 
It leads to a bit of code duplication.

Reply:
Well they are not the same list. I guess I could replace both the procCsvs and unprocCSVs by a single map with a boolean as a value, iterate that then call either processObject or cleanObject. Thoughts ?

Reply:
Or maybe you could have a helper function, such as 
```
func parallelApply(list s3.ListObjectsOutput, fn func(o types.Object))  {
	var wg sync.WaitGroup
	cDone := make(chan bool)
	wg.Add(len(list))
	for _, item := range list {
		o := item
		go func() {
			defer wg.Done()
			fn(o)
		}()
	}

	wg.Wait()
}
```

And call it like so : 
```
	if len(unprocCsvs.Contents) > 0 {
		parallelApply(unprocCsvs, func(o types.Object) error {
			if processObject(ctx, dwn, upl, influxWriter, o); err != nil {
				log.Error().Err(err).Msg("Processing error")
			}
		})
	}

	if opts.CleanObjects && len(procCsvs.Contents) > 0 {
		parallelApply(procCsvs, func(o types.Object) error {
			if cleanObject(ctx, s3Cli, o); err != nil {
				log.Error().Err(err).Msg("Cleaning error")
			}
		})
	}
```